source/application/AMDSCentralServerSGMScaler.cpp

```diff
@@ -107,12 +116,47 @@ void AMDSCentralServerSGMScaler::onServerChangedToDwellState(int index){
 	emit serverChangedToDwellState(index);
 }
 
+#include "ClientRequest/AMDSClientStartTimeToEndTimeDataRequest.h"
 void AMDSCentralServerSGMScaler::onNewScalerScanDataReceivedd(const AMDSDataHolderList &scalerScanCountsDataHolder)
 {
 	AMDSThreadedBufferGroup * bufferGroup = bufferGroupManagers_.value(scalerConfigurationMap_->scalerName());
 	if (bufferGroup) {
 		bufferGroup->append(scalerScanCountsDataHolder);
+
+		if(internalRequestActive_){
+			internalRequestActive_ = false;
+
+			QDateTime endTime = QDateTime::currentDateTime();
+			quint64 asMsecs = quint64(dwellSecondsRequested_*1000);
+			QDateTime startTime = endTime.addMSecs(-asMsecs);
+			AMDSClientStartTimeToEndTimeDataRequest *localRequest = new AMDSClientStartTimeToEndTimeDataRequest(AMDSClientRequest::Binary, endTime, "", scalerConfigurationMap_->scalerName(), false, true, startTime, endTime, bufferGroup->bufferGroupInfo(), this);
+			bufferGroup->bufferGroup()->processClientRequest(localRequest, true);
+		}
 	} else {
 		AMDSRunTimeSupport::debugMessage(AMDSRunTimeSupport::ErrorMsg, this, AMDS_SGM_SERVER_ALT_INVALID_BUFFERGROUP_NAME, QString("Failed to find bufferGroup for %1").arg(scalerConfigurationMap_->scalerName()));
 	}
 }
+
+void AMDSCentralServerSGMScaler::onScalerDetectorRequestFlattenedData(double seconds)
+{
+	internalRequestActive_ = true;
+	dwellSecondsRequested_ = seconds;
+}
+
+#include "DataHolder/AMDSScalarDataHolder.h"
+void AMDSCentralServerSGMScaler::onInternalRequestProcessed(AMDSClientRequest *clientRequest)
+{
+//	qDebug() << "Internal request was just processed " << clientRequest->metaObject()->className();
```

Comment:
I was going to leave these here for the time being. I expect we might change types, data types, or request structures in the near future and these debugs acted as a really solid place to check things as they flew by.